File: sion_fasion_shop/ProyectoMVC/SionFashioWebApplicationMVC/SionFashioWebApplicationMVC/Models/paise.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace SionFashioWebApplicationMVC.Models;

public partial class paise
{
    [Key]
    public int id_pais { get; set; }

    [Required(ErrorMessage = "El nombre del país es obligatorio.")]
    [StringLength(255, ErrorMessage = "El nombre del país no puede exceder los 255 caracteres.")]
    public string nombre_pais { get; set; } = null!;

    [InverseProperty("id_paisNavigation")]
    public virtual ICollection<departamento> departamentos { get; set; } = new List<departamento>();
}

File: sion_fasion_shop/Entregables/Trimestre V/Sistema de inventario en Node.Js/Bck-End SionFashion-Shop/models/paise.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

// Esquema del modelo Paise
const paiseSchema = new Schema({
    nombre_pais: {
        type: String,
        required: [true, 'El nombre del país es obligatorio.'],
        maxlength: [255, 'El nombre del país no puede exceder los 255 caracteres.']
    },
    departamentos: [{
        type: Schema.Types.ObjectId,
        ref: 'departamento' // Asegúrate de tener el modelo "Departamento" definido
    }]
});

// Crea el modelo Paise
const Paise = mongoose.model('Paise', paiseSchema);

module.exports = Paise;
